.travis.yml

```diff
@@ -1,23 +1,47 @@
-language: go
+dist: trusty
+sudo: required
 
+# define the dependence env
+language: go
 os:
   - linux
-
 go:
   - "1.13"
-
+services:
+  - docker
 env:
   - GO111MODULE=on
-
 install: true
 
+# define ci-stage
 script:
+  # license-check
+  - echo 'start license check'
   - go fmt ./... && [[ -z `git status -s` ]]
   - sh before_validate_license.sh
   - chmod u+x /tmp/tools/license/license-header-checker
   - /tmp/tools/license/license-header-checker -v -a -r -i vendor  /tmp/tools/license/license.txt . go  && [[ -z `git status -s` ]]
+  # unit-test
+  - echo 'start unit-test'
   - chmod u+x before_ut.sh && ./before_ut.sh
   - go mod vendor && go test ./... -coverprofile=coverage.txt -covermode=atomic
+  # integrate-test
+  - echo 'start integrate-test'
+  # start zookeeper registry insecure listen in [:]:2181
+  - docker run -d --network host zookeeper
+  - ROOTDIR=$(pwd)
+  - cd ./test/integrate/dubbo/go-client && docker build . -t  ci-consumer --build-arg PR_ORIGIN_REPO=${TRAVIS_PULL_REQUEST_SLUG} --build-arg PR_ORIGIN_COMMITID=${TRAVIS_PULL_REQUEST_SHA} && cd $ROOTDIR
+  - cd ./test/integrate/dubbo/go-server && docker build . -t  ci-provider --build-arg PR_ORIGIN_REPO=${TRAVIS_PULL_REQUEST_SLUG} --build-arg PR_ORIGIN_COMMITID=${TRAVIS_PULL_REQUEST_SHA} && cd $ROOTDIR
+  - docker run -d --network host ci-provider
+  - docker run -it --network host ci-consumer

```

Comment:
`docker run -d --network host ci-provider` run container in background. when `ci-consumer` started, `ci-provider` may be not boot completed.

Reply:
if ci-provider not boot completed, the ci flow will faild, zookeeper container have sample problem. but our registry will retry.